config/v3_3_experimental/types/file.go

```diff
@@ -30,7 +30,8 @@ func (f File) Validate(c path.ContextPath) (r report.Report) {
 }
 
 func (f File) validateOverwrite() error {
-	if util.IsTrue(f.Overwrite) && f.Contents.Source == nil {
+	sources := f.Contents.GetSources()
+	if util.IsTrue(f.Overwrite) && len(sources) == 0 {
```

Comment:
nit: let's drop the `sources` variable assignment and in-line it in the `if` as it isn't re-used.